File: 0198-house-robber/0198-house-robber.java
class Solution {
    public int rob(int[] nums) {
        int r1 = 0, r2 = 0;

        for (int num: nums) {
            int temp = Math.max(r1 + num, r2);
            r1 = r2;
            r2 = temp;
        }

        return Math.max(r1, r2);
    }
}

File: 0198-house-robber/0198-house-robber.cpp
class Solution {
public:
    int rob(vector<int>& nums) {
        int first = 0;
        int second = 0;

        for (int i = 0; i < nums.size(); i++) {
            int temp = max(first + nums[i], second);
            first = second;
            second = temp;
        }

        return second;
    }
};